app/src/main/java/org/simple/clinic/overdue/communication/Communication.kt

```diff
@@ -0,0 +1,72 @@
+package org.simple.clinic.overdue.communication
+
+import android.arch.persistence.room.Dao
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.OnConflictStrategy
+import android.arch.persistence.room.PrimaryKey
+import android.arch.persistence.room.Query
+import com.squareup.moshi.Json
+import io.reactivex.Flowable
+import org.simple.clinic.patient.SyncStatus
+import org.simple.clinic.util.RoomEnumTypeConverter
+import org.threeten.bp.Instant
+import java.util.UUID
+
+@Entity(tableName = "Communication")
+data class Communication(
+    @PrimaryKey val id: UUID,
+    val appointmentId: UUID,
+    val userId: UUID,
+    val type: Type,
+    val result: Result,
+    val syncStatus: SyncStatus,
+    val createdAt: Instant,
+    val updatedAt: Instant
+) {
+
+  enum class Type {
+    @Json(name = "manual_call")
+    MANUAL_CALL,
+
+    @Json(name = "voip_call")
+    VOIP_CALL;
+
+    class RoomTypeConverter : RoomEnumTypeConverter<Type>(Type::class.java)
+  }
+
+  enum class Result {
+    @Json(name = "unavailable")
+    UNAVAILABLE,
+
+    @Json(name = "agreed_to_visit")
+    AGREED_TO_VISIT,
+
+    @Json(name = "denied_to_visit")
+    DENIED_TO_VISIT;
+
+    class RoomTypeConverter : RoomEnumTypeConverter<Result>(Result::class.java)
+  }
+
+  @Dao
+  interface RoomDao {
+
+    @Query("SELECT * FROM Communication WHERE syncStatus = :status")
+    fun withSyncStatus(status: SyncStatus): Flowable<List<Communication>>
+
+    @Query("UPDATE Communication SET syncStatus = :to WHERE syncStatus = :from")
+    fun updateSyncStatus(from: SyncStatus, to: SyncStatus)
+
+    @Query("UPDATE Communication SET syncStatus = :to WHERE id IN (:ids)")
+    fun updateSyncStatus(ids: List<UUID>, to: SyncStatus)
+
+    @Query("SELECT * FROM Communication WHERE id = :id LIMIT 1")
+    fun getOne(id: UUID): Communication?
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun save(schedules: List<Communication>)
+
+    @Query("SELECT COUNT(*) FROM Communication")
```

Comment:
Can we use just the primary key column when getting a count of rows?

Reply:
Aye